go/tasks/plugins/k8s/kfoperators/tensorflow/tensorflow.go

```diff
@@ -0,0 +1,248 @@
+		return nil, flyteerr.Errorf(flyteerr.BadTaskSpecification, "nil task specification")
+	}
+
+	tensorflowTaskExtraArgs := plugins.DistributedTensorflowTrainingTask{}
+	err = utils.UnmarshalStruct(taskTemplate.GetCustom(), &tensorflowTaskExtraArgs)
+	if err != nil {
+		return nil, flyteerr.Errorf(flyteerr.BadTaskSpecification, "invalid TaskSpecification [%v], Err: [%v]", taskTemplate.GetCustom(), err.Error())
+	}
+
+	podSpec, err := flytek8s.ToK8sPodSpec(ctx, taskCtx.TaskExecutionMetadata(), taskCtx.TaskReader(), taskCtx.InputReader(), taskCtx.OutputWriter())
+	if err != nil {
+		return nil, flyteerr.Errorf(flyteerr.BadTaskSpecification, "Unable to create pod spec: [%v]", err.Error())
+	}
+
+	overrideDefaultContainerName(taskCtx, podSpec)
+
+	workers := tensorflowTaskExtraArgs.GetWorkers()
+	psReplicas := tensorflowTaskExtraArgs.GetPsReplicas()
+	chiefReplicas := tensorflowTaskExtraArgs.GetChiefReplicas()
+
+	jobSpec := tfOp.TFJobSpec{
+		TTLSecondsAfterFinished: nil,
+		TFReplicaSpecs: map[tfOp.TFReplicaType]*commonOp.ReplicaSpec{
+			tfOp.TFReplicaTypePS: {
+				Replicas: &psReplicas,
+				Template: v1.PodTemplateSpec{
+					Spec: *podSpec,
+				},
+				RestartPolicy: commonOp.RestartPolicyNever,
+			},
+			tfOp.TFReplicaTypeChief: {
+				Replicas: &chiefReplicas,
+				Template: v1.PodTemplateSpec{
+					Spec: *podSpec,
+				},
+				RestartPolicy: commonOp.RestartPolicyNever,
+			},
+			tfOp.TFReplicaTypeWorker: {
+				Replicas: &workers,
+				Template: v1.PodTemplateSpec{
+					Spec: *podSpec,
+				},
+				RestartPolicy: commonOp.RestartPolicyNever,
+			},
+		},
+	}
+
+	job := &tfOp.TFJob{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       tfOp.Kind,
+			APIVersion: tfOp.SchemeGroupVersion.String(),
+		},
+		Spec: jobSpec,
+	}
+
+	return job, nil
+}
+
+// Analyses the k8s resource and reports the status as TaskPhase. This call is expected to be relatively fast,
+// any operations that might take a long time (limits are configured system-wide) should be offloaded to the
+// background.
+func (tensorflowOperatorResourceHandler) GetTaskPhase(ctx context.Context, pluginContext k8s.PluginContext, resource k8s.Resource) (pluginsCore.PhaseInfo, error) {
+	app := resource.(*tfOp.TFJob)
+
+	workersCount := app.Spec.TFReplicaSpecs[tfOp.TFReplicaTypeWorker].Replicas
+	psReplicasCount := app.Spec.TFReplicaSpecs[tfOp.TFReplicaTypePS].Replicas
+	chiefCount := app.Spec.TFReplicaSpecs[tfOp.TFReplicaTypeChief].Replicas
+
+	taskLogs, err := getLogs(app, *workersCount, *psReplicasCount, *chiefCount)
+	if err != nil {
+		return pluginsCore.PhaseInfoUndefined, err
+	}
+
+	currentCondition, err := extractCurrentCondition(app.Status.Conditions)
+	if err != nil {
+		return pluginsCore.PhaseInfoUndefined, err
+	}
+
+	occurredAt := time.Now()
+	statusDetails, _ := utils.MarshalObjToStruct(app.Status)
+	taskPhaseInfo := pluginsCore.TaskInfo{
+		Logs:       taskLogs,
+		OccurredAt: &occurredAt,
+		CustomInfo: statusDetails,
+	}
+
+	switch currentCondition.Type {
```

Comment:
can we share this code between all kf operators like - 
pytorch and tensorflow. they seem to be the same?